File: Solutions/682. Baseball Game/682. Baseball Game.go
func calPoints(ops []string) int {
    var stk []int
    for _,o:=range ops{
        if o=="C"{
            stk=stk[:len(stk)-1]
        }else if o=="D"{
            stk=append(stk,stk[len(stk)-1]*2)
        }else if o=="+"{
            x:=stk[len(stk)-1];stk=stk[:len(stk)-1]
            y:=stk[len(stk)-1]
            stk=append(stk,x)
            stk=append(stk,x+y)
        }else{
            i,err:=strconv.Atoi(o)
            if err==nil{
                stk=append(stk,i)
            }
        }
    }
    res:=0
    for _,s:=range stk{
        res+=s
    }
    return res
}

File: Solutions/682. Baseball Game/682. Baseball Game.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        vector<int> stk;
        for(int i=0;i<ops.size();++i){
            if(ops[i]=="C"){
                stk.pop_back();
            }else if(ops[i]=="D"){
                stk.push_back(stk.back()*2);
            }else if(ops[i]=="+"){
                int x=stk.back();stk.pop_back();
                int y=stk.back();
                stk.push_back(x);
                stk.push_back(x+y);
            }else{
                stk.push_back(stoi(ops[i]));
            }
        }
        return accumulate(stk.begin(),stk.end(),0);
    }
};